be/src/formats/parquet/encoding_dict.h

```diff
@@ -203,22 +205,93 @@ class DictDecoder<Slice> final : public Decoder {
         return Status::OK();
     }
 
-    Status get_dict_values(const std::vector<int32_t>& dict_codes, Column* column) override {
+    Status get_dict_values(const std::vector<int32_t>& dict_codes, const NullableColumn& nulls,
+                           Column* column) override {
+        const std::vector<uint8_t>& null_data = nulls.immutable_null_column_data();
+        bool has_null = nulls.has_null();
+        bool all_null = false;
+
+        if (has_null) {
+            size_t count = SIMD::count_nonzero(null_data);
+            all_null = (count == null_data.size());
+        }
+
+        // dict codes size and column size HAVE TO BE EXACTLY SAME.
         std::vector<Slice> slices(dict_codes.size());
-        for (size_t i = 0; i < dict_codes.size(); i++) {
-            slices[i] = _dict[dict_codes[i]];
+        if (!has_null) {
+            for (size_t i = 0; i < dict_codes.size(); i++) {
+                slices[i] = _dict[dict_codes[i]];
+            }
+        } else if (!all_null) {
+            size_t size = dict_codes.size();
+            const uint8_t* null_data_ptr = null_data.data();
+
+            size_t i = 0;
+            for (i = 0; (i + 8) < size; i += 8) {
+                uint64_t partial_all_null;
+                memcpy(&partial_all_null, null_data_ptr, 8);
+                if (partial_all_null != 0) {
+                    for (int j = 0; j < 8; j++) {
+                        if (!null_data_ptr[j]) {
+                            slices[i + j] = _dict[dict_codes[i + j]];
```

Comment:
may be partial_all_not_null is better, so that we don't need if on every element

Reply:
yes, the name should be `partial_all_not_null`.